src/modules/auth/auth.service.ts

```diff
@@ -19,4 +22,24 @@ export class AuthService {
     user.password = undefined;
     return user;
   }
+
+  public async verifyEmail(token: string) {
+    try {
+      // Verification token
+      const decoded = (await jwt.verify(
+        token,
+        config.jwt.secret!,
+      )) as JwtPayload;
+      // Check if user still exist
+      const user = await this.userService.getUserById(decoded.id);
+      if (user) {
+        user.isEmailConfirmed = true;
+        await user.save();
+      }
+    } catch (error) {
```

Comment:
Throwing error in this case is fine.
In fact it's required otherwise response will be successful.